app/presenters/licences/licence-contacts.presenter.js

```diff
@@ -0,0 +1,61 @@
+'use strict'
+
+/**
+ * Formats data for the `/licences/{id}/contact-details` view licence contact details page
+ * @module ViewLicenceContactDetailsPresenter
+ */
+
+/**
+ * Formats data for the `/licences/{id}/contact-details` view licence contact details page
+ *
+ * @returns {Object} The data formatted for the view template
+ */
+function go (contacts) {
+  return {
+    customerId: _findCustomerId(contacts),
+    licenceContacts: _licenceContacts(contacts)
+  }
+}
+
+function _findCustomerId (contacts) {
+  const customerContact = contacts.find((contact) => {
+    return contact.communicationType === 'Licence Holder'
+  })
+
+  if (customerContact) {
+    return customerContact.companyId
+  }
+
+  return null
+}
+
+function _licenceContactName (contact) {
+  if (contact.contactId) {
```

Comment:
This logic does not match the previous changes. Both First AND last name should be present. 

This is the old UI code: 

```
     <td class="govuk-table__cell" scope="row">
                        {% if contact.firstName and contact.lastName %}
                            {{ contact.firstName }} {{contact.lastName}}
                        {% else %}
                            {{ contact.name }}
                        {% endif %}
                      </td>
```

Reply:
Contact name being the company name.

Reply:
So, I believe in the legacy UI code 'contact' is not a direct mapping to `crm_v2.contacts`. Its mappers etc will decorate the entity with stuff you don't get from the table. `name` doesn't exist as a column in the table. In fact, we have an equivalent on the model (because it is more complicated than it first appears! https://github.com/DEFRA/water-abstraction-system/blob/main/app/models/contact.model.js#L78 )

When we've dug into the legacy code regards showing a 'contact' the general principle seems to be if `crm_v2.contact` is present it takes precedence. But if you check the database `last_name` is never null but `first_name` sometimes is. Hence not assuming it will be there and having a test for if it's not.